nims-notes-things/src/Form.js

```diff
@@ -23,56 +24,77 @@ const Form = (props) => {
     setFormData(newForm);
   };
 
-  const handleOnBlur = (e, key) => {
-    empty.map((data) => {
-      return <h2>key.data</h2>;
+  const handleOnBlur = (value, key) => {
+    setFormEmpty((previousState) => {
+      return {
+        ...previousState,
+        [key]: value === "",
+      };
     });
   };
 
-  // const handleOnBlurName = e => {
-  //   if(formData.name === ''){
-  //     alert("This field is required")
-  //   }
-  // }
-  // const handleOnBlurDate = e => {
-  //   if(formData.name === ''){
-  //     alert("This field is required")
-  //   }
-  // }
-  // const handleOnBlurNote = (e,key) => {
-  //   if(formData.name === ''){
-  //     alert("This field is required")
-  //   }
-  // }
   return (
-    <div className="App">
-      <Input
-        name="name"
-        placeholder="Please enter Your name"
-        onChange={(e) => handleOnChange(e, "name")}
-        onBlur={(e) => handleOnBlur(e.target.value, "name")}
-        value={formData.name}
-      />
-      <h5>Name: {formData.name}</h5>
+    <form>
+      <div>
+        <InnerForm>
+          {empty.name && <ErrorMessage>Name field is required</ErrorMessage>}
+          <Input
+            name="name"
+            placeholder="Please enter Your name"
+            onChange={(e) => handleOnChange(e, "name")}
+            onBlur={(e) => handleOnBlur(e.target.value, "name")}
+            value={formData.name}
+          />
+          <ErrorMessage>
```

Comment:
no need to wrap the error message here...
`{empty.date && <ErrorMessage>Date field is required</ErrorMessage>}` is enough